binding-hs/src/Client.hs

```diff
@@ -0,0 +1,192 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE TypeOperators         #-}
+
+module Client
+  ( envCreate
+  , envListAll
+  , envReset
+  , envStep
+  , envActionSpaceInfo
+  , envActionSpaceSample
+  , envActionSpaceContains
+  , envObservationSpaceInfo
+  , envMonitorStart
+  , envMonitorClose
+  , envClose
+  , upload
+  , shutdownServer
+  , EnvID (..)
+  , InstID (..)
+  , Environment (..)
+  , Observation (..)
+  , Step (..)
+  , Outcome (..)
+  , Info (..)
+  , Action (..)
+  , Monitor (..)
+  , Config (..)
+  ) where
+
+import           Data.Aeson          (FromJSON, Object (..), ToJSON)
+import qualified Data.HashMap.Strict as HM (HashMap (..))
+import           Data.Proxy          (Proxy (..))
+import qualified Data.Text           as T (Text)
+import           GHC.Generics
+import           Network.HTTP.Client (Manager, defaultManagerSettings,
+                                      newManager)
+import           Servant.API
+import           Servant.Client      (BaseUrl (..), ClientM, client)
+import           Servant.HTML.Lucid  (HTML)
+
+--------------------------------------------------------------------------------
+-- | Data types and instances
+
+newtype EnvID = EnvID { env_id :: T.Text }
```

Comment:
I agree. Here are some interesting slides on this matter: http://www.slideshare.net/oxbow_lakes/age-is-not-an-int
Even though I loved this talk and swore to follow these practices, I still forget to write my own types sometimes.

Thanks for pointing it out. :)